Trappist/src/Promact.Trappist.Repository/Questions/QuestionRepository.cs

```diff
@@ -111,24 +116,49 @@ public QuestionRepository(TrappistDbContext dbContext)
             return languageNameList;
         }
 
-
-        public async Task UpdateCodeSnippetQuestionAsync(int questionId, QuestionAC questionAC, string userId)
+        public async Task UpdateCodeSnippetQuestionAsync(QuestionAC questionAC, string userId)
         {
-            var updatedQuestion = await _dbContext.Question.FindAsync(questionId);
-            var updatedCodeSnippetQuestion = await _dbContext.CodeSnippetQuestion.FindAsync(questionId);
+            var updatedQuestion = await _dbContext.Question.FindAsync(questionAC.Question.Id);
 
-            Mapper.Map(questionAC.Question, updatedQuestion);
-            Mapper.Map(questionAC.CodeSnippetQuestion, updatedCodeSnippetQuestion);
+            await _dbContext.Entry(updatedQuestion).Reference(x => x.CodeSnippetQuestion).LoadAsync();
+            await _dbContext.Entry(updatedQuestion.CodeSnippetQuestion).Collection(x => x.CodeSnippetQuestionTestCases).LoadAsync();
+
+            var testCases = updatedQuestion.CodeSnippetQuestion.CodeSnippetQuestionTestCases;
+            var updatedTestCase = questionAC.CodeSnippetQuestion.CodeSnippetQuestionTestCases;
 
+            Mapper.Map(questionAC.Question, updatedQuestion);
+            Mapper.Map(questionAC.CodeSnippetQuestion, updatedQuestion.CodeSnippetQuestion);
             updatedQuestion.UpdatedByUserId = userId;
-            updatedCodeSnippetQuestion.Question = updatedQuestion;
+            await _dbContext.SaveChangesAsync();
 
             using (var transaction = _dbContext.Database.BeginTransaction())
             {
-                _dbContext.Question.Update(updatedQuestion);
+                //Handling one to many relationship with TestCase
+                //Finding TestCase to be updated, deleted and added
+                var testCaseToUpdate = updatedTestCase.Where(x => testCases.Any(y => y.Id == x.Id)).ToList();
+                var testCaseToDelete = testCases.Where(x => !updatedTestCase.Any(y => y.Id == x.Id)).ToList();
+                var testCaseToAdd = updatedTestCase.Where(x => !testCases.Any(y => y.Id == x.Id)).ToList();
+
+                //Deleting TestCases
+                _dbContext.CodeSnippetQuestionTestCases.RemoveRange(testCaseToDelete);
+                await _dbContext.SaveChangesAsync();
+
+                //Adding TestCases
+                testCaseToAdd.ForEach(x =>
+                {
+                    x.CodeSnippetQuestionId = updatedQuestion.CodeSnippetQuestion.Id;
+                    x.Id = 0;
+                });
+                await _dbContext.CodeSnippetQuestionTestCases.AddRangeAsync(testCaseToAdd);
                 await _dbContext.SaveChangesAsync();
 
-                _dbContext.CodeSnippetQuestion.Update(updatedCodeSnippetQuestion);
+                //Updating TestCases
+                testCaseToUpdate.ForEach(x =>
+                {
+                    var testCase = testCases.First(test => test.Id == x.Id);
```

Comment:
If there is only one record with that id then please use single instead of first